packages/vite/src/vite-node.ts

```diff
@@ -48,7 +48,7 @@ export function viteNodePlugin (ctx: ViteBuildContext): VitePlugin {
           markInvalidates(server.moduleGraph.getModulesByFile(typeof plugin === 'string' ? plugin : plugin.src))
         }
         for (const template of ctx.nuxt.options.build.templates) {
-          markInvalidates(server.moduleGraph.getModulesByFile(template?.src))
+          markInvalidates(server.moduleGraph.getModulesByFile(template?.src as string))
```

Comment:
because `ctx.nuxt.options.build.templates` is no longer `any`, TypeScript shows an error here, because the `src` in [NuxtTemplate](https://github.com/nuxt/nuxt/blob/main/packages/schema/src/types/nuxt.ts#L30) can be `undefined`. 

I'm guilty of not trying out what happens (or even can it happen) if it is `undefined` here, but I can investigate it further.

Reply:
Hm. We do need to investigate - is it possible that the module would be added by `template.dst` instead? 🤔  This may be the origin of a bug we're encountering elsewhere...

Reply:
By quickly looking at the code, I think we expect to use `template.getContents` when `src` is not present ([see](https://github.com/nuxt/nuxt/blob/main/packages/kit/src/template.ts#L54)), and we call that function [at one place in the codebase](https://github.com/nuxt/nuxt/blob/main/packages/kit/src/internal/template.ts#L21), but we never set `src` to anything if it is `undefined`, so we can potentially call `server.moduleGraph.getModulesByFile()` with `undefined`. 

Although yeah, I just glanced over the code, I'm gonna check this out more thoroughly tomorrow.